### PR TITLE
Move file related models to pkg/model/file

### PR DESCRIPTION
**Pull request by Adirio**
# Description

Last movement of model packages that were inside `pgk/scaffold` to `pkg/model`.

- Move model.File to its own subpackage
- Change input.File name to input.Template to avoid collisions with file.File
- Move pkg/scaffold/input into pkg/model/file
- Delete unused tests
- Resolve naming conflicts
- Stop injecting ProjectPath, BoilerplatePath and Version into templates
    BoilerplatePath is being used once in v1 and once in v2, and the rest are never being used, so injeceting them is not justified.
- Inject from universe instead of from scaffold
- Remove unused fields from Scaffold
- Remove validate function as was only used once and has 3 LoC

# Motivation

This PR is scoped under #1218.

### CODE REVIEW

**Comment by Adirio** [comment]
[Travis error](https://travis-ci.org/kubernetes-sigs/kubebuilder/jobs/644267966#L331-L351) makes no sense at all. It is writting `-` as the boilerplate and that is only done when `model.WithoutBoilerplate` is specified, which is not the case for that file.

Changed `-` to `// No copyright` for `model.WithoutBoilerplate` to verify it comes from there.

File: pkg/scaffold/v1/manager/apis.go [hunk 1]
```diff
@@ -21,15 +21,16 @@ import (
...
-var _ input.File = &APIs{}
+var _ file.Template = &APIs{}
 
 // APIs scaffolds a apis.go to register types with a Scheme
 type APIs struct {
-	input.Input
-
+	file.Input
+	// BoilerplatePath is the path to the boilerplate file
+	BoilerplatePath string
```
Comment (mengqiy) [inline pkg/scaffold/v1/manager/apis.go]: Is this used anywhere?

File: pkg/scaffold/scaffold.go [hunk 2]
```diff
@@ -77,92 +61,10 @@ type Plugin interface {
...
-			b.SetDomain(s.Config.Domain)
-		}
-		if b, ok := t.(input.Version); ok {
-			b.SetVersion(s.Config.Version)
-		}
-		if b, ok := t.(input.Repo); ok {
-			b.SetRepo(s.Config.Repo)
-		}
-		if b, ok := t.(input.ProjecPath); ok {
-			b.SetProjectPath(s.ConfigPath)
```
[the remark this answers is not in the document]
Comment (Adirio) [reply pkg/scaffold/scaffold.go]: None of the templates required the path of the configuration file. Two of them require the path of the boilerplate, one to compute a relative path in v1 and the makefile in v2.

File: pkg/scaffold/v2/makefile.go [hunk 3]
```diff
@@ -17,30 +17,32 @@ limitations under the License.
...
-var _ input.File = &Makefile{}
+var _ file.Template = &Makefile{}
 
 // Makefile scaffolds the Makefile
 type Makefile struct {
-	input.Input
+	file.Input
 	// Image is controller manager image name
 	Image string
+	// BoilerplatePath is the path to the boilerplate file
```
[the remark this answers is not in the document]
Comment (Adirio) [reply pkg/scaffold/v2/makefile.go]: It uses it for controller-gen (around line 104):

```
# Generate code
generate: controller-gen
	$(CONTROLLER_GEN) object:headerFile={{printf "%q" .BoilerplatePath}} paths="./..."
```

File: pkg/scaffold/v1/manager/apis.go (hunk 1 above)
Comment (Adirio) [reply pkg/scaffold/v1/manager/apis.go]: A couple lines below to compute a relative path.

**Pull request merged**